src/alembic/versions/20251016_1000_7e5b5cf430ff_destructive_migration_filesystem_and_.py

```
@@ -0,0 +1,178 @@
+    # Create a named snapshot before destructive changes
+    # This snapshot can be used to restore the database if needed
+    logger.warning("⚠️  DESTRUCTIVE MIGRATION: Creating database snapshot before migration...")
+    from program.utils.cli import snapshot_database
+    from pathlib import Path
+
+    snapshot_dir = Path("./data/db_snapshot")
+    snapshot_name = "pre_destructive_migration_834cba7d26b4"
+
+    if snapshot_database(snapshot_dir, snapshot_name):
+        logger.log("DATABASE", f"✅ Database snapshot created: {snapshot_dir / (snapshot_name + '.sql')}")
+        logger.log("DATABASE", f"   To restore: alembic downgrade -1")
```

Ruff (0.14.0)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)